src/main/java/com/google/step/snippet/servlets/UserSettingServlet.java

```diff
@@ -24,11 +29,36 @@
 /** Servelet that renders user setting page */
 @WebServlet("/user")
 public class UserSettingServlet extends HttpServlet {
+  private static final String WEBSITE_PARAMETER = "website";
+  private static final String LANGUAGE_PARAMETER = "language";
+  private static final String ID_PARAMETER = "id";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     // Forward the request to the template (which is a servlet itself).
     request.getRequestDispatcher("WEB-INF/templates/user_dashboard.jsp").forward(request, response);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/home");
```

Comment:
users should be redirected back to the page that they were previously on? not necessarily homepage

Reply:
fixed

Reply:
actually I realized, shouldn't this just always redirect to /user since we know the post request will only happen on the /user page?